events/api.py

```diff
@@ -8,6 +12,20 @@ class Meta:
         model = Event
         fields = '__all__'
 
+    def validate(self, data):
+        start_time = data.get('start_time', None)
+        end_time = data.get('end_time', None)
+
+        # PATCH updates only 'state', so check that start and end times are present in the data
+        if (start_time and end_time) and (start_time > end_time):
+            raise serializers.ValidationError(_('Event must start before ending.'))
+
+        # TODO should the 7 day minimum waiting time be configurable from e.g. admin?
+        # discussion can be had in https://github.com/City-of-Helsinki/haravajarjestelma/issues/8
+        if start_time and (start_time < timezone.now() + timedelta(days=7)):
```

Comment:
IMO it would be better if the value was configurable through the Django settings file and an env variable.

Also, I think this should take into account only days and not times, for example this week Wednesday at 12:00, the earliest you should be able to add an event is at next week Thursday 00:00, not at next week Wednesday 12:01.

Reply:
Good point. Changed validation and tests to capture this approach.